WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2305,7 +2305,7 @@ private GutenbergPropsBuilder getGutenbergPropsBuilder() {
         String wpcomLocaleSlug = languageString.replace("_", "-").toLowerCase(Locale.ENGLISH);
 
         // If this.mIsXPostsCapable has not been set, default to allowing xPosts
-        boolean enableXPosts = mIsXPostsCapable == null || mIsXPostsCapable;
+        boolean enableXPosts = mSite.isUsingWpComRestApi() && (mIsXPostsCapable == null || mIsXPostsCapable);
```

Comment:
Mmm, interesting. Asking here since I don't see a mention in the PR itself: what's the value of `mIsXPostsCapable` in the case of self-hosted sites? Is it `null` perhaps? If that's the case, I feel like we also need to update the line comment to align with the updated conditional, wdyt @SiobhyB, perhaps in a separate PR? Thanks!

Reply:
@hypest, good thought! I didn't have time to get to this today, but will tomorrow. :)

Reply:
Thanks for the follow up @SiobhyB !